### PR TITLE
Added inference and a way to retrieve the "File Hints" for assistants

### PR DESCRIPTION
**Pull request by phact**
This PR allows you to create an `assistant` and run inference against a row from the benchmark.

For https://github.com/raymyers/swe-bench-util/pull/6 to work we needed to be able to get back what files are being used under the hood by Assistants inside a `run`. OpenAI's Assistants API has a placeholder for this but they don't use it. 

From the OpenAI [docs](https://platform.openai.com/docs/api-reference/assistants-streaming/run-step-delta-object):
image

I implemented this functionality in astra-assitants (v0.1.9) and released it https://github.com/datastax/astra-assistants-api/pull/15 

To test just run: 

    python -m swe_bench_util index astra-assistants

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -158,3 +158,4 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 .idea/
+checkouts/
```
Comment (raymyers) [inline .gitignore:161]: Good catch

File: swe_bench_util/index/astra_assistants.py [hunk 2]
```diff
@@ -30,10 +33,9 @@ def upload_file(file_path) -> str | None:
                 "rb",
             ),
             purpose="assistants",
-            embedding_model="text-embedding-3-large",
```
Comment (raymyers) [inline swe_bench_util/index/astra_assistants.py:33]: Just curious, does the embedding model default without this?

**Comment by raymyers** [comment]
Good stuff, merging. Fixed a small merge conflict that was due to formatting, in the future `ruff format` should help avoid that. If there's a way that makes it harder to forget to run that would be neat.

File: swe_bench_util/index/astra_assistants.py (hunk 2 above)
Comment (phact) [reply swe_bench_util/index/astra_assistants.py:33]: I think default is ada-002. We should get scientific and test out a few out including different providers.

**Pull request merged**